### ISSUE TITLE
Increase report performance

### ISSUE THREAD

**Issue by twelch** [seasketch/geoprocessing#159]
Need:
- Hitting the limits of what can be computed by a single maxed-out Lambda (Bermuda
- Wait time for reports to return is on the order of 2-3 minutes in some cases.  And for some large designs, never return (timeout)

Goals
* support data complexity NZ required with large vector habitat dataset (we had to downsample the data)
* Support higher resolution rasters, such as from EmLab for input prioritization (BioOracle, etc.)
* Run Allen Coral Atlas report (% by coral type, vector) for a large EEZ (Maldives)
* Run GEBCO depth report for a large EEZ (Maldives)

Requirement:
* analysis output should be the same for the same input.  If an exception should be considered, let's discuss.

Possibilities:
- [ ] SeaSketch should execute reports early.  Requires (#22) to be resolved
- [x] Ensure local lru report result cache is working as expected
- [x] Break up calculations into pieces and run them in parallel.  Most gp functions loop through multiple datasources or data classes.
- [x] Speed up toolbox functions and underlying dependencies (turf, geoblaze)
  - #155
  - #132
  - Switch the clipping in vector overlap toolbox functions from Turf to [JSTS](https://github.com/bjornharrtell/jsts) or potentially [geos-wasm](https://github.com/chrispahm/geos-wasm)
  - Switch the clipping in vector overlap toolbox functions from Turf to DuckDB, which can handle loading of datasource from flatgeobuf in S3, and performing the vector operations against sketch.

Tasks
- [ ] Add performance benchmarking of functions from the start and strategically move the needle

**Comment by twelch** [comment]
@DanielJDufour I'd like to hear more about what precomputing tile statistics means to you, direct discussion probably best for this.  In the way that I think about it, the question becomes how to use precomputed tiles in practice when a feature overlay is done.  Note the requirement I added up top that the output should be the same for any method used to improve performance. 

I'm reminded that I created this related ticket some time ago - https://github.com/seasketch/geoprocessing/issues/89 that consider some ideas.  I kind of like your idea of using geotiff's directly if it allows it to work with whatever projection the raster data is in.
